contracts/cw-ics721-bridge/src/contract.rs

```diff
@@ -0,0 +1,567 @@
+        .add_messages(messages))
+}
+
+fn execute_mint(
+    deps: Deps,
+    env: Env,
+    info: MessageInfo,
+    class_id: String,
+    token_ids: Vec<String>,
+    token_uris: Vec<String>,
+    receiver: String,
+) -> Result<Response, ContractError> {
+    if info.sender != env.contract.address {
+        return Err(ContractError::Unauthorized {});
+    }
+    if token_ids.len() != token_uris.len() {
+        return Err(ContractError::ImbalancedTokenInfo {});
+    }
+
+    let receiver = deps.api.addr_validate(&receiver)?;
+    let cw721_addr = CLASS_ID_TO_NFT_CONTRACT.load(deps.storage, class_id)?;
+
+    let mint_messages = token_ids
+        .into_iter()
+        // Can zip without worrying about dropping data as we assert
+        // that lengths are the same above.
+        .zip(token_uris.into_iter())
+        .map(|(token_id, token_uri)| -> StdResult<WasmMsg> {
+            let msg = cw721_base::msg::ExecuteMsg::<Empty, Empty>::Mint(cw721_base::MintMsg {
+                token_id,
+                token_uri: Some(token_uri),
+                owner: receiver.to_string(),
+                extension: Empty::default(),
+            });
+            Ok(WasmMsg::Execute {
+                contract_addr: cw721_addr.to_string(),
+                msg: to_binary(&msg)?,
+                funds: vec![],
+            })
+        })
+        .collect::<StdResult<Vec<_>>>()?;
+
+    Ok(Response::default()
+        .add_attribute("method", "execute_mint")
+        .add_messages(mint_messages))
+}
+
+#[allow(clippy::too_many_arguments)]
+fn execute_do_instantiate_and_mint(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    class_id: String,
+    class_uri: Option<String>,
+    token_ids: Vec<String>,
+    token_uris: Vec<String>,
+    receiver: String,
+) -> Result<Response, ContractError> {
+    if info.sender != env.contract.address {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Optionally, instantiate a new cw721 contract if one does not
+    // yet exist.
+    let submessages = if CLASS_ID_TO_NFT_CONTRACT.has(deps.storage, class_id.clone()) {
+        vec![]
+    } else {
+        let message = SubMsg::<Empty>::reply_on_success(
+            WasmMsg::Instantiate {
+                admin: None,
+                code_id: CW721_CODE_ID.load(deps.storage)?,
+                msg: to_binary(&cw721_base::msg::InstantiateMsg {
+                    // Name of the collection MUST be class_id as this is how
+                    // we create a map entry on reply.
+                    name: class_id.to_string(),
+                    symbol: class_id.to_string(),
+                    minter: env.contract.address.to_string(),
+                })?,
+                funds: vec![],
+                // Attempting to fit the class ID in the label field
+                // can make this field too long which causes weird
+                // data errors in the SDK.
+                label: "ICS771 backing CW721".to_string(),
```

Comment:
Can we just do an abbreviation/cut off here? At least we can get some information about what is in the contract rather than this static string which ends up being meaningless.

Reply:
i get that this is less informative than it could be. honestly, i'd prefer to leave this as is. the class ID is currently stored in both the name and symbol so it is on chain and quite visible.

reason for preferring leaving as-is: the errror you get for having too long of a label is quite opaque. it is not clear to me that chains have the same length limits, nor is it clear to me that they won't change. in addition, slicing strings is actually [kinda hard](https://doc.rust-lang.org/nightly/book/ch08-02-strings.html#slicing-strings) and dangerous.